### ISSUE TITLE
Remove 'Accessing low memory' warning

### ISSUE THREAD

**Issue by Liniya** [z00m128/sjasmplus#92]
Hello. I would like to suggest you to get rid of this message:

`warning: Accessing low memory address 0x009B, is it ok?: ld a,(#009B)`

To be more specific, I think the 'less warnings' option wouldn't cut it here, this is literally an unnecessary warning.

Even on the Spectrum, it is very common to test the ROM data for some purpose, such as detecting specific models (e.g. differentiating between Sinclair vs Amstrad 128K), or a peripheral ROM version, etc. Some Spectrum peripherals return data by intercepting accesses to specific 'ROM' low memory ranges, and so on.

On other Z80 systems such as SMS, arcade machines etc, this memory range is often available for user data and code, so it makes even less sense.

**Comment by ped7g** [comment]
how you can avoid the warning (any of these):
1) in the source `OPT --syntax=m` or `--syntax=m` on command line
2) add end-of-line comment starting with "ok" like: `ld a,(13) ; ok`
3) use brackets to make it clear you are accessing memory: `ld a,[13]`

I had this discussion with other sjasmplus contributors, and most of them do not like it and want it to be removed too.

But so far from the anecdotal evidence it seems to work well (the huge MSX project I was helping to migrate to sjasmplus did show the warning three times - from total ~460k of source lines - BTW it's using only single memory address in the 0..255 range for reading, even if the platform is well known to use this region starting at $0000 as RAM and bios/user values, 99% of them are beyond address 255).

On my own sources I had so far zero false positives, only real problems were reported twice.

If you can be more specific how often you run into this, and how often it is just false positive, and how often it helps to catch bugs, it would be better, but I'm taking this as +1 for false positives, and will keep an eye on it. Maybe I will flip the "m" option from "disable" to "enable", especially on the v1.x branch, as it's new feature.

**Comment by Liniya** [comment]
I see, so this is only for addresses 0~255? Is it perhaps intended as a measure against mistyping IN as LD, or something along the line?

Personally, I do not often run into a situation where there is a lower memory access, and it is unintended. In my case, it is mostly about detecting Spectrum models, so I assume that would make 1, maybe 2 per project. I may have mistyped IN as LD once or twice in a couple of years or so. :) Nothing too disastrous there either, and as a general rule, it is easily found.

Oh, and making it disabled by default would be just as good by me, of course.

**Comment by ped7g** [comment]
it's for me forgetting the round parentheses are memory access...
I sometimes do as in x86 asm + trauma from C macros: `ld a,(flag1|flag2)` just to be sure the value is correctly calculated, but then I get memory access by accident, and the correct Zilog syntax for that is `ld a,+(flag1|flag2)` or no parentheses at all.

So it's only about 0..255 values, as 256+ is not correct "8b immediate" any way, so you either want really access memory, or you have two bugs in single instruction. IIRC in the MSX project I added `; ok ; original comment` to all three of them... the "; ok" will become probably a generic way to suppress any warning in v2.x, currently works in v1.14.3 to suppress this low memory access and fake instruction warning.

Keep me up to date if you are running into it often, so I have some idea, but for the moment I find it very low-frequency and the ";ok" comment seems to me reasonable mechanics to confirm the assembler that you know what you are doing.

For a seasoned ZX dev doing only Z80 asm this is probably not an issue, but for people switching between platforms and different syntax I think this guardian will do either nothing, or be better than nothing, so for the moment I'm still keeping it on by default. But I hear you. Thanks for the feedback.

**Comment by ped7g** [comment]
At this moment it seems the amount of issues with this warning is still extremely low.

In my own sources it did help me since then zero times (it had one opportunity, but I managed to do the double-bug both using memory access by accident, and calculating value larger than 255, so the warning was missing and I found the bug in debugger).

So I'm going to close this for the moment, but any comments with some stats about how many projects you have and how many warnings you had to suppress vs how many times it did help you to find bug are still welcome (with enough evidence this may get eventually disabled by default, I'm still open to this variant, I just believe current setup is causing more good than harm).
